addons/account/models/account_bank_statement.py

```diff
@@ -178,10 +178,38 @@ def _compute_journal_id(self):
 
     @api.depends('balance_end_real', 'balance_end')
     def _compute_is_complete(self):
+        incomplete_statements = self.filtered(lambda stmt: not stmt.is_complete)
+
         for stmt in self:
             stmt.is_complete = stmt.line_ids and stmt.currency_id.compare_amounts(
                 stmt.balance_end, stmt.balance_end_real) == 0
 
+        # Process the bank statement report.
+        complete_stmts = incomplete_statements.filtered('is_complete')
+        if self.ids and complete_stmts:
+            IrActionsReport = self.env['ir.actions.report'].sudo()
```

Comment:
rather call it ir_actions_report_sudo . By repeating the sudo in the name, you make sure people don't forget it when calling the record.